app/src/main/java/com/nginx/image/core/PhotoResizer.java

```diff
@@ -247,65 +262,43 @@ private String makeJson(ConcurrentHashMap<String,String> imagesURLMap) {
      *
      * @return an {@link ImageInformation} object
      */
-    private ImageInformation resize(File resizedImageFile, int maxSize, BufferedImage originalBuffImage) {
-        ImageInformation imageData = new ImageInformation(1,0,0);
+    private ImageInformation resize(File resizedImageFile, int maxSize, BufferedImage originalBuffImage) throws Exception {
         try {
 
             double scale;
 
             // maxSize == -1 means that the original size should be used
-            if(maxSize == -1) {
-                imageData = new ImageInformation(1, originalBuffImage.getWidth(), originalBuffImage.getHeight());
-                return imageData;
-            } else if (originalBuffImage.getWidth() > originalBuffImage.getHeight()) {
-                // if the image is wider than it is tall, then scale base on width
-                scale = (double) maxSize/originalBuffImage.getWidth();
-            } else {
-                // if the image is taller than it is wide, then scale base on height
-                scale = (double) maxSize/originalBuffImage.getHeight();
+            if(maxSize != -1) {
+                if (originalBuffImage.getWidth() > originalBuffImage.getHeight()) {
+                    // if the image is wider than it is tall, then scale base on width
+                    scale = (double) maxSize/originalBuffImage.getWidth();
+                } else {
+                    // if the image is taller than it is wide, then scale base on height
+                    scale = (double) maxSize/originalBuffImage.getHeight();
+                }
+                // get the scaling dimensions
+                int widthScale = (int) round(scale * originalBuffImage.getWidth());
+                int heightScale = (int) round(scale * originalBuffImage.getHeight());
+
+                BufferedImage resizedBuffImage;
+                resizedBuffImage = new BufferedImage(widthScale, heightScale, BufferedImage.TYPE_INT_RGB);
+                Image tmp = originalBuffImage.getScaledInstance(widthScale, heightScale, BufferedImage.SCALE_SMOOTH);
+                resizedBuffImage.getGraphics().drawImage(tmp, 0, 0, null);
+                originalBuffImage = resizedBuffImage;
+                resizedBuffImage.flush();
+                tmp.flush();
             }
+            ImageInformation imageData = new ImageInformation(1, originalBuffImage.getWidth(), originalBuffImage.getHeight());
+            writeJpg(resizedImageFile, originalBuffImage, compressionQuality);
```

Comment:
If we call `config.getResizerConfiguration().getCompressionQuality()` here, then we don't need the class property `compressionQuality` and it's consistent with the calls to `config.getImageSizeConfiguration().getImageSize(size)` paradigm of making method calls.

Reply:
updated.